Z64Utils/Forms/ObjectAnalyzerForm.cs

```diff
@@ -244,19 +244,44 @@ private void listView_map_SelectedIndexChanged(object sender, EventArgs e)
                         StringWriter sw = new StringWriter();
                         for (int n = 0; n < matrices.Matrices.Count; n++)
                         {
-                            sw.WriteLine($" ┌                                                ┐ ");
+                            string[,] mtxStr = new string[4, 4];
+                            int[] columnsWidth = { 0, 0, 0, 0 };
+                            for (int i = 0; i < 4; i++)
+                            {
+                                for (int j = 0; j < 4; j++)
+                                {
+                                    // see Matrix_MtxToMtxF

```

Comment:
The matrix `ToMatrix4` returns is transposed, hm.

```
 ┌                                             ┐ 
 │   0.13165283    -0.9912567             0  0 │ 
 │ -0.057769775  -0.007659912   -0.99827576  0 │ 
 │     0.989563    0.13142395  -0.058288574  0 │ 
 │          600          -100            50  1 │ 
 └                                             ┘ 
```

The only current usage of `ToMatrix4` is here
https://github.com/Random06457/Z64Utils/blob/dev/Z64Utils/F3DZEX/Render/Renderer.cs#L748

which seems to do funky stuff:
```
                            //mtxf = curMtx * mtxf;
                            mtxf *= RdpMtxStack.Top();
```
this does `mtxf = mtxf * RdpMtxStack.Top();` right, not the opposite

so is there an issue with `ToMatrix4` or do I roll with how it currently behaves and index it with `[j,i]` instead of `[i,j]`

@Random06457

Reply:
`Matrix4` is an OpenTK struct which represents matrics in column-major order for historical reason. (See https://github.com/opentk/opentk/issues/512).
I'd suggest just using `[j, i]` since it's not really an issue. (You could change it to row-major order but you'd have to modify some skeleton/shader related code.)